### PR TITLE
[API] Sites endpoint

### PR DESCRIPTION
**Pull request by xlecours**
API 0.0.4 dev
Returns the list of sites for an /api/v0.0.0.4-dev/sites GET request.

This is a rebased version of https://github.com/aces/Loris/pull/7466

### CODE REVIEW

File: modules/api/docs/LorisRESTAPI_v0.0.4-dev.md [hunk 1]
```diff
@@ -1122,3 +1122,33 @@ Returns raw file with the appropriate MimeType headers for the archival file
...
+      "Name": "string",
+      "Alias": "string",
+      "MRI alias": "string"
+    },
+    ...
+  ]
+}
+```
+
+`Alias` and `MRI alias` are short strings that are used as "tags" to identify a site or a group of sites. Those aliases are often used for display or file nameing purposes. (e.g: PSCID generation, minc files naming convention, ...).
```
Comment (christinerogers) [inline modules/api/docs/LorisRESTAPI_v0.0.4-dev.md:1151]:
```suggestion
`Alias` and `MRI alias` are short strings that are used as "tags" to identify a site or a group of sites. Those aliases are often used for display or file naming purposes. (e.g: PSCID generation `MTL00001`). The `MRI alias` field is typically populated only for sites which collect imaging data. 
```

File: modules/api/docs/LorisRESTAPI_v0.0.4-dev.md [hunk 2]
```diff
@@ -1122,3 +1122,33 @@ Returns raw file with the appropriate MimeType headers for the archival file
...
+
+* Note that only the `Name` property is unique accross all sites.
```
Comment (christinerogers) [inline modules/api/docs/LorisRESTAPI_v0.0.4-dev.md:1153]:
```suggestion
* Note that only the `Name` property is unique across all sites.
```
typo

**Review by christinerogers — CHANGES_REQUESTED** [review]
Hi Xavier, I've reviewed the alias/mri_alias part and can commit my suggested changes if you grant permission -- thanks. 
Any other parts of this PR I should look at in particular?

**Review by christinerogers — APPROVED** [review]

**Comment by xlecours** [comment]
@christinerogers Thanks for the corrections. Since you mentioned that some clients required this endpoint, can you tell if the current response structure and content is satisfying?

**Comment by christinerogers** [comment]
> @christinerogers Thanks for the corrections. Since you mentioned that some clients required this endpoint, can you tell if the current response structure and content is satisfying?

Hi Xavier, given the `getSites` call returns the same information and response structure, nothing will break.  I took a look and I don't think anything changed regarding this particular response, true?  given I read that right, I give my 👍

File: modules/api/static/schema-v0.0.4-dev.yml [hunk 3]
```diff
@@ -44,6 +44,18 @@ paths:
...
         '204':
           description: The visit stage was updated successfully
+  /sites:
+    get:
+      tags:
+        - Sites
+      summary: List all sites
+      responses:
+        '200':
+          description: A list of all sites; regardless of projects and/or user's permissions.
```
Comment (jesscall) [inline modules/api/static/schema-v0.0.4-dev.yml:54]: Should this be updated to reflect that there is a [userSiteMatch](https://github.com/aces/Loris/pull/7771/files#diff-53f7d851ca4412ecf8f287a9c194d02e267ef9e2705f7923a1f458308e565567R113) filter ?
```suggestion
          description: A list of all sites for the requesting user.
```

**Review by maltheism — CHANGES_REQUESTED** [review]
Conflict with Changelog

**Comment by laemtl** [comment]
@xlecours I need that feature for the Wizard. Any chance that can be part of the next release? Let me know if I can help.

**Pull request merged**